### PR TITLE
Remove unnecessary registry supplier redirects

### PR DESCRIPTION
**Pull request by embeddedt**
These are redundant now that the registry rework and registration order adjustments have been completed.

This PR is an updated version of https://github.com/neoforged/NeoForge/pull/520 for 20.5, please look at the discussions in that PR for more context.

### CODE REVIEW

File: patches/net/minecraft/world/item/BucketItem.java.patch [hunk 1]
```diff
@@ -100,11 +75,12 @@
...
 +
-+    private final java.util.function.Supplier<? extends Fluid> fluidSupplier;
-+    public Fluid getFluid() { return fluidSupplier.get(); }
-+
 +    protected boolean canBlockContainFluid(@Nullable Player player, Level worldIn, BlockPos posIn, BlockState blockstate)
 +    {
 +        return blockstate.getBlock() instanceof LiquidBlockContainer && ((LiquidBlockContainer)blockstate.getBlock()).canPlaceLiquid(player, worldIn, posIn, blockstate, this.content);
++    }
++
++    public Fluid getFluid() {
```
Comment (Technici4n) [inline patches/net/minecraft/world/item/BucketItem.java.patch:83]: alternatively we could AT the content

File: patches/net/minecraft/world/level/block/StairBlock.java.patch [hunk 2]
```diff
@@ -1,63 +0,0 @@
...
-@@ -84,7 +_,12 @@
-         return voxelshape;
-     }
- 
-+    @Deprecated // Forge: Use the other constructor that takes a Supplier
-     public StairBlock(BlockState p_56862_, BlockBehaviour.Properties p_56863_) {
-+        this(() -> p_56862_, p_56863_);
-+    }
-+
-+    public StairBlock(java.util.function.Supplier<net.minecraft.world.level.block.state.BlockState> p_56862_, BlockBehaviour.Properties p_56863_) {
```
Comment (Technici4n) [inline patches/net/minecraft/world/level/block/StairBlock.java.patch:29]: Isn't this one still potentially useful?

**Review by Technici4n — CHANGES_REQUESTED** [review]

File: patches/net/minecraft/world/level/block/StairBlock.java.patch (hunk 2 above)
Comment (embeddedt) [reply patches/net/minecraft/world/level/block/StairBlock.java.patch:29]: I don't necessarily think so, modders can just register their content in an appropriate order. There shouldn't ever be a case where there is a cyclic dependency between a stair and non-stair block IMO.

Comment (Technici4n) [reply patches/net/minecraft/world/level/block/StairBlock.java.patch:29]: @XFactHD you mentioned something about these, right?

**Review by XFactHD — APPROVED** [review]

**Comment by Technici4n** [comment]
:shipit:

File: patches/net/minecraft/world/item/BucketItem.java.patch (hunk 1 above)
Comment (Technici4n) [reply patches/net/minecraft/world/item/BucketItem.java.patch:83]: Keeping the method is fine and will at least not be a breaking change.

**Review by Technici4n — APPROVED** [review]

**Pull request merged**